Source/Controllers/SPAppController.m

```diff
@@ -2027,6 +2039,32 @@ - (IBAction)reloadBundles:(id)sender
 
 }
 
+- (void)removeBundle:(NSString*)bundle{
+	
+	NSString *bundlePath = [fileManager applicationSupportDirectoryForSubDirectory:SPBundleSupportFolder error:nil];
+	
+	NSString *thePath = [NSString stringWithFormat:@"%@/%@", bundlePath, bundle];
+	
+	SPLog(@"the path %@", thePath);
+	
+	if([fileManager fileExistsAtPath:thePath isDirectory:nil]) {
+		NSError *error = nil;
+		
+		[fileManager removeItemAtPath:thePath error:&error];
+		
+		if(error != nil) {
+			SPLog(@"file could not be deleted: %@", thePath);
+		}
+		else{
+			SPLog(@"file was deleted: %@", thePath);
+			[badBundles addObject:bundle];
+		}
+	}
+	else{
+		SPLog(@"file does not exist %@", thePath);
+	}
```

Comment:
As a style thing, but this could be a good candidate for applying early-return (no else) programming paradigms. Instead of checking if the file exists and then having an else, the method could check if the file doesn't exist with an early return and otherwise delete the file?


Something like
```
NSString *bundlePath = [fileManager applicationSupportDirectoryForSubDirectory:SPBundleSupportFolder error:nil];

NSString *thePath = [NSString stringWithFormat:@"%@/%@", bundlePath, bundle];

SPLog(@"the path %@", thePath);

if(![fileManager fileExistsAtPath:thePath isDirectory:nil]) {
    SPLog(@"file does not exist %@", thePath);
    return;
}

NSError *error = nil;

[fileManager removeItemAtPath:thePath error:&error];

if(error != nil) {
    SPLog(@"file could not be deleted: %@", thePath);
    return;
}

SPLog(@"file was deleted: %@", thePath);
[badBundles addObject:bundle];
```

I think it's a little easier to follow, no nested Ifs. The TL/DR of the methodology is the normal, successful flow of logic should be from the start to the very end of the method, with early return conditions breaking out when a specific pre or post condition is not properly met. Just a style comment!